### PR TITLE
tests/driver_at86rf2xx: fix var lengths and unused vars

### PR DESCRIPTION
**Pull request by kYc0o**
While compiling tests/driver_at86rf2xx with clang in OS X it discovers several minor issues.

This PR fixes them.

### CODE REVIEW

File: tests/driver_at86rf2xx/cmd.c [hunk 1]
```diff
@@ -248,7 +249,7 @@ static int send(int iface, le_uint16_t dst_pan, uint8_t *dst, size_t dst_len,
 {
     int res;
     netdev_ieee802154_t *dev;
-    const size_t count = 2;         /* mhr + payload */
+    size_t count = 2;         /* mhr + payload */
```
Comment (gebart) [inline tests/driver_at86rf2xx/cmd.c]: How about static const instead?

Comment (kYc0o) [reply tests/driver_at86rf2xx/cmd.c]: Well clang was complaining about the `const` so that's why I changed, since IMHO there's not necessary. What `static` would change?

**Comment by kYc0o** [comment]
Well, cppcheck complains about something I was noticed already, but have no clue how it should be solved. Any ideas?

File: tests/driver_at86rf2xx/cmd.c [hunk 2]
```diff
@@ -148,7 +148,8 @@ int txtsnd(int argc, char **argv)
 {
     char *text;
     uint8_t addr[_MAX_ADDR_LEN];
-    int iface, idx = 2, res;
+    int iface, idx = 2;
+    size_t res;
```
Comment (gebart) [inline tests/driver_at86rf2xx/cmd.c]: Why did you need to change this? The parse function returns int

File: tests/driver_at86rf2xx/cmd.c (hunk 1 above)
Comment (gebart) [reply tests/driver_at86rf2xx/cmd.c]: Static const means it won't take up stack space

Comment (kYc0o) [reply tests/driver_at86rf2xx/cmd.c]: Well I get the original error if I change it:

```console
/Users/facosta/git/RIOT-OS/RIOT/tests/driver_at86rf2xx/cmd.c:253:25: error: variable length array folded to constant array as an extension
      [-Werror,-Wgnu-folding-constant]
    struct iovec vector[count];
                        ^~~~~
1 error generated.
```

File: tests/driver_at86rf2xx/cmd.c (hunk 2 above)
Comment (kYc0o) [reply tests/driver_at86rf2xx/cmd.c]: Because of this:

```console
/Users/facosta/git/RIOT-OS/RIOT/tests/driver_at86rf2xx/cmd.c:160:36: error: comparison of integers of different signs: 'int' and
      'unsigned long' [-Werror,-Wsign-compare]
            if ((res <= 0) || (res > sizeof(pan))) {
                               ~~~ ^ ~~~~~~~~~~~
1 error generated.
```

Comment (miri64) [reply tests/driver_at86rf2xx/cmd.c]: It would always make more sense to me to change the comparison than, instead of the type.

File: tests/driver_at86rf2xx/cmd.c (hunk 1 above)
Comment (miri64) [reply tests/driver_at86rf2xx/cmd.c]: Can you make it a macro constant then instead?

```C
#define MAC_VECTOR_SIZE  (2)
```

Doesn't make much sense to use stack space for that constant anyway.

File: tests/driver_at86rf2xx/recv.c [hunk 3]
```diff
@@ -98,7 +98,7 @@ void recv(netdev_t *dev)
     printf("Seq.: %u\n", (unsigned)ieee802154_get_seq(buffer));
     od_hex_dump(buffer + mhr_len, data_len - mhr_len, 0);
     printf("txt: ");
-    for (int i = mhr_len; i < data_len; i++) {
+    for (size_t i = mhr_len; i < data_len; i++) {
```
Comment (miri64) [inline tests/driver_at86rf2xx/recv.c]: Use rather `unsigned` as a counting variable.

File: tests/driver_at86rf2xx/cmd.c (hunk 2 above)
Comment (miri64) [reply tests/driver_at86rf2xx/cmd.c]: Oh I just sah that `send()` below expects type `size_t` anyway (see remove of cast), so changing the type to `size_t` makes sense in the end, I think.

File: tests/driver_at86rf2xx/recv.c (hunk 3 above)
Comment (kYc0o) [reply tests/driver_at86rf2xx/recv.c]: The change here is because the comparison shouldn't be made between two variables of different size. `mhr_len` and `data_len` are declared as `size_t`.

**Comment by kYc0o** [comment]
> Since you changed the type of res to size_t (which is an unsigned type) it doesn't make sense to check if it is res is less than 0. That is what cppcheck is complaining about. I guess it wasn't a good idea to change that type after all, so let's go with casting to unsigned for the comparisons clang is complaining about.

Then I think we're doing something wrong here, the compiler is stating the obvious, so I'd prefer to make the right comparisons rather than trick the compiler to avoid the warning.

File: tests/driver_at86rf2xx/cmd.c (hunk 2 above)
Comment (kYc0o) [reply tests/driver_at86rf2xx/cmd.c]: Agree.

**Comment by miri64** [comment]
> Then I think we're doing something wrong here, the compiler is stating the obvious, so I'd prefer to make the right comparisons rather than trick the compiler to avoid the warning.

if a signed value is on the side that is supposed to be smaller, (e.g. `res < size`) than it is a *right* comparison. The range of an unsigned is always larger of its respective signed type.

File: tests/driver_at86rf2xx/recv.c (hunk 3 above)
Comment (miri64) [reply tests/driver_at86rf2xx/recv.c]: a) where did you get that from? Comparisons between variables of different size shouldn't be a problem at all
b) I think on all our platforms `sizeof(size_t) == sizeof(unsigned)` is true.

Comment (kYc0o) [reply tests/driver_at86rf2xx/recv.c]: > a) where did you get that from? Comparisons between variables of different size shouldn't be a problem at all

I said they _shouldn't_, because it _might_ lead to errors, if a counter variable is smaller than the variable it's compared with of course it cannot reach the required length. It will only work if the count doesn't exceed it's own type length (which yes, it's mostly what happens).

> b) I think on all our platforms sizeof(size_t) == sizeof(unsigned) is true.

I _think_ it's not enough, our code needs to be portable. `sizeof()` returns `size_t`, you can find [here](https://gcc.gnu.org/ml/gcc-help/2006-06/msg00188.html) a small discussion about it.

**Comment by kYc0o** [comment]
> if a signed value is on the side that is supposed to be smaller, (e.g. res < size) than it is a right comparison. The range of an unsigned is always larger of its respective signed type.

I agree, but why not just do what's correct? Just compare two variables of the same length and no compiler will complain.

**Comment by miri64** [comment]
> I agree, but why not just do what's correct? Just compare two variables of the same length and no compiler will complain.

Because it wastes memory unnecessarily. Comparing two values of different sizes is still correct if you know what you are doing.

**Comment by kYc0o** [comment]
> Because it wastes memory unnecessarily. Comparing two values of different sizes is still correct if you know what you are doing.

This _optimisation_ trick might lead to errors. I prefer to waste 2B for a correct type and length than "expect" we all know what we're doing. I believe compiler warnings are not thrown just for fun...

**Comment by miri64** [comment]
> This optimisation trick might lead to errors. I prefer to waste 2B for a correct type and length than "expect" we all know what we're doing. I believe compiler warnings are not thrown just for fun...

It's not a trick. But you are changing types instead of operation which *definitely* leads to errors.

**Comment by miri64** [comment]
(see murdock output)
